de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/util/WindowStateStorageWrapper.java

```diff
@@ -0,0 +1,155 @@
+package org.iplantc.de.desktop.client.presenter.util;
+
+import org.iplantc.de.client.models.UserInfo;
+import org.iplantc.de.client.models.WindowState;
+import org.iplantc.de.client.util.WebStorageUtil;
+import org.iplantc.de.desktop.client.views.windows.WindowBase;
+
+import com.google.common.base.Strings;
+import com.google.gwt.core.client.GWT;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A wrapper to store and retrieve window state from web local storage
+ *
+ * Created by sriram on 1/23/18.
+ */
+public class WindowStateStorageWrapper {
+    public static final String LOCAL_STORAGE_PREFIX = "de.";
+
+    UserInfo userInfo;
+    String windowType;
+    String tag;
+
+    String prefix;
+    String suffix;
+    WindowBase.WindowStateFactory wsf = GWT.create(WindowBase.WindowStateFactory.class);
+
+    public WindowStateStorageWrapper(UserInfo userInfo, String windowType, String tag) {
+        this.userInfo = userInfo;
+        this.windowType = windowType;
+        this.tag = tag;
+        prefix = WindowStateStorageWrapper.LOCAL_STORAGE_PREFIX + windowType;
+        suffix = "#" + tag + "#" + userInfo.getUsername();
+        this.wsf = wsf;
+    }
+
+
+    /**
+     * Save window state to web local storage
+     *
+     * @param ws
+     */
+    public void saveWindowState(WindowState ws) {
+        if (ws != null) {
+            saveHeight(ws.getHeight());
+            saveWidth(ws.getWidth());
+            saveLeft(ws.getWinLeft());
+            saveTop(ws.getWinTop());
+            saveMaximizedState(ws.isMaximized());
+            saveMiniMinimizedState(ws.isMinimized());
+            storeMap(ws.getAdditionalWindowStates());
+        }
+    }
+
+    /**
+     * Retrieve window state from web local storage
+     *
+     * @return WindowState object
+     */
+    public WindowState retrieveWindowState() {
+        WindowState ws = wsf.windowState().as();
+        ws.setMaximized(retrieveMaximizedState());
+        ws.setMinimized(retrieveMinimizedState());
+        ws.setWinTop(retrieveTop());
+        ws.setWinLeft(retrieveLeft());
+        ws.setWidth(retrieveWidth());
+        ws.setHeight(retrieveHeight());
+        ws.setTag(tag);
+        ws.setAdditionalWindowStates(retrieveMap());
+        return ws;
+    }
+
+    void saveHeight(String height) {
+        WebStorageUtil.writeToStorage(prefix + WindowState.HEIGHT + suffix, height + "");
```

Comment:
I like the refactor! I think it's much cleaner and organized to have everything regarding read/write in one class. 🌟 

The last thing I think it might need is a single method for dealing with the keys.  Right now, if we need to add something to the window preferences that gets saved, we still have to make sure we're constructing the key correctly.  Or, if we need to change/rearrange how the key is constructed we have to change it in several places in this class. 

Maybe something like
`WebStorageUtil.writeToStorage(getKey(WindowState.HEIGHT), height)` or
`write(WindowState.HEIGHT, height)`
and then something similar for read?

Reply:
Sounds good. Let me make the change.